src/dialog/index.ts

```diff
@@ -176,37 +190,44 @@ function showDialog(options?: IDialogOptions): Promise<IButtonItem> {
   options.host = host;
   options.body = options.body || '';
   okButton.text = options.okText ? options.okText : 'OK';
-  let buttons = options.buttons || [cancelButton, okButton];
-  let buttonNodes = buttons.map(createButton);
-  let dialog = createDialog(options, buttonNodes);
-  host.appendChild(dialog);
-  // Focus the ok button if given.
-  let index = buttons.indexOf(okButton);
-  if (index !== -1) {
-    buttonNodes[index].focus();
-  }
-  return new Promise<IButtonItem>((resolve, reject) => {
-    buttonNodes.map(node => {
-      node.addEventListener('click', evt => {
-        if (node.contains(evt.target as HTMLElement)) {
+  let buttons = options.buttons = options.buttons || [cancelButton, okButton];
+  if (options.body instanceof Widget) {
```

Comment:
I think that this logic could be combined, since a `Widget` can take a node in its constructor, we should pass either a node a widget to the `WidgetDialog` class and let it either use the widget or create a widget to wrap the node.